MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp

```diff
@@ -255,40 +155,12 @@ void QtReflMainView::on_actionSearch_triggered() {
 }
 
 /**
-This slot notifies the presenter that the "transfer" button has been pressed
+This slot requests the list of runs to transfer to ReflMainViewPresenter
+Then updates m_tableView
 */
 void QtReflMainView::on_actionTransfer_triggered() {
-  m_presenter->notify(IReflPresenter::TransferFlag);
-}
-
-/**
-This slot notifies the presenter that the "export table" button has been pressed
-*/
-void QtReflMainView::on_actionExportTable_triggered() {
-  m_presenter->notify(IReflPresenter::ExportTableFlag);
-}
-
-/**
-This slot notifies the presenter that the "import table" button has been pressed
-*/
-void QtReflMainView::on_actionImportTable_triggered() {
-  m_presenter->notify(IReflPresenter::ImportTableFlag);
-}
-
-/** This slot is used to syncrhonise the two instrument selection widgets */
-void QtReflMainView::on_comboProcessInstrument_currentIndexChanged(int index) {
-  ui.comboSearchInstrument->setCurrentIndex(index);
-  m_calculator->setCurrentInstrumentName(
-      ui.comboProcessInstrument->currentText().toStdString());
-  m_calculator->processInstrumentHasBeenChanged();
-}
-
-/** This slot is used to syncrhonise the two instrument selection widgets */
-void QtReflMainView::on_comboSearchInstrument_currentIndexChanged(int index) {
-  ui.comboProcessInstrument->setCurrentIndex(index);
-  m_calculator->setCurrentInstrumentName(
-      ui.comboProcessInstrument->currentText().toStdString());
-  m_calculator->processInstrumentHasBeenChanged();
+  auto runs = m_presenter->getRunsToTransfer();
+  m_tableView->transferRuns(runs);
```

Comment:
I've added back the `IReflPresenter`'s flag `TransferFlag`. This way, when the user clicks on `Transfer` the view notifies the presenter. The presenter then gets the selected search rows and sends them to `m_tablePresenter` via the interface `IReflTablePresenter`. This way `QtReflMainView` does not need to act on `m_tableView` directly.